mautrix_instagram/portal.py

```diff
@@ -1691,14 +1700,35 @@ def _get_thread_name(self, thread: Thread) -> str:
                 return tpl.format(
                     displayname=ui.full_name or ui.username, id=ui.pk, username=ui.username
                 )
-            pass
         elif thread.thread_title:
             return self.config["bridge.group_chat_name_template"].format(name=thread.thread_title)
-        else:
-            return ""
+
+        return ""
+
+    async def _get_thread_avatar(self, source: u.User, thread: Thread) -> Optional[ContentURI]:
+        if self.is_direct or not thread.thread_image:
+            return None
+        best: Optional[ThreadImageCandidate] = None
+        for candidate in thread.thread_image.image_versions2.candidates:
+            if best is None or candidate.width > best.width:
+                best = candidate
+        if not best:
+            return None
+        data, mimetype = await self._download_instagram_file(source, best.url)
+        if not data:
+            return None
+        mxc = await self.main_intent.upload_media(
+            data=data,
+            mime_type=mimetype,
+            filename=thread.thread_image.id,
+            async_upload=self.config["homeserver.async_media"],
+        )
+        return mxc
 
     async def update_info(self, thread: Thread, source: u.User) -> None:
         changed = await self._update_name(self._get_thread_name(thread))
+        if thread_avatar := await self._get_thread_avatar(source, thread):
+            changed = await self._update_photo(thread_avatar)
```

Comment:
This looks like it reuploads the avatar on each sync, which isn't optimal. Other bridges store a remote avatar ID and compare that to detect if it should be updated. The `puppet` table also has `photo_id` for that, although that's a bit hacky (it parses the URL), here it looks like `ThreadImage` already has an `id` field that could be used?

Reply:
Ah, I'll can add `avatar_id` to the `Portal`